docker-compose.yml

```diff
@@ -51,6 +63,7 @@ services:
       - DB_MIN_CONN_SIZE=1
       - DB_MAX_CONN_SIZE=1
       - USE_API_HYDRATE=${USE_API_HYDRATE:-false}
+      - UVICORN_ROOT_PATH=/api/v1/pgstac
```

Comment:
People might find a bit complicated to have to use the `nginx` service

you could `in theory` do something like 👇 
```yml
  nginx-sqlalchemy:
    extends:
      service: app-sqlalchemy
    ports:
    environment:
      - UVICORN_ROOT_PATH=/api/v1/sqlalchemy

  nginx:
    image: nginx
    ports:
      - 80:80
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - [app](nginx-sqlalchemy:)
    command: ["nginx-debug", "-g", "daemon off;"]
```

Reply:
oh, I don't think you can `extend` services that have already `depends`.

> Keep in mind that volumes_from and depends_on are never shared between services using extends. These exceptions exist to avoid implicit dependencies; you always define volumes_from locally. This ensures dependencies between services are clearly visible when reading the current file. Defining these locally also ensures that changes to the referenced file don’t break anything.

you may need to re-add this

Reply:
Good point, this does break the links on the existing apps when going around nginx.  I think maybe we should just add a second compose stack - `docker-compose-nginx-proxy.yml`.

Reply:
@vincentsarago I picked this up and implemented as an optional docker-compose override file. I re-requested your review -- I tested locally but could use a second set of eyes.

Reply:
I think @geospatial-jeff would need to 👍 this too

Reply:
@geospatial-jeff ping -- if you don't mind taking a look, we can clear this PR out before 2.4.4. Thanks!